realm/realm-library/src/main/java/io/realm/internal/ColumnInfo.java

```diff
@@ -165,22 +167,36 @@ public void copyFrom(ColumnInfo src) {
             throw new NullPointerException("Attempt to copy null ColumnInfo");
         }
 
-        indicesMap.clear();
-        indicesMap.putAll(src.indicesMap);
+        indicesFromJavaFieldNames.clear();
+        indicesFromJavaFieldNames.putAll(src.indicesFromJavaFieldNames);
+        indicesFromColumnNames.clear();
+        indicesFromColumnNames.putAll(src.indicesFromColumnNames);
         copy(src, this);
     }
 
     @Override
     public String toString() {
         StringBuilder buf = new StringBuilder("ColumnInfo[");
-        buf.append(mutable).append(",");
-        if (indicesMap != null) {
+        buf.append("mutable="+mutable).append(",");
+        if (indicesFromJavaFieldNames != null) {
+            buf.append("JavaFieldNames=[");
             boolean commaNeeded = false;
-            for (Map.Entry<String, ColumnDetails> entry : indicesMap.entrySet()) {
+            for (Map.Entry<String, ColumnDetails> entry : indicesFromJavaFieldNames.entrySet()) {
                 if (commaNeeded) { buf.append(","); }
                 buf.append(entry.getKey()).append("->").append(entry.getValue());
                 commaNeeded = true;
             }
+            buf.append("]");
+        }
+        if (indicesFromColumnNames != null) {
+            buf.append(", InternalFieldNames=[");
+            boolean commaNeeded = false;
+            for (Map.Entry<String, ColumnDetails> entry : indicesFromColumnNames.entrySet()) {
+                if (commaNeeded) { buf.append(","); }
```

Comment:
`commaNeeded = true;` should be inside the if?

Reply:
No, in that case, it would never be set to the value `true`.